<a href="https://colab.research.google.com/github/santiagoprado12/Titanic---Machine-Learning-from-Disaster/blob/main/Proyecto_Final_Titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Subir archivos**

In [36]:
train<-read.csv("train.csv",stringsAsFactors = FALSE)
test<-read.csv("test.csv",stringsAsFactors = FALSE)
str(train)
str(test)

'data.frame':	891 obs. of  12 variables:
 $ PassengerId: int  1 2 3 4 5 6 7 8 9 10 ...
 $ Survived   : int  0 1 1 1 0 0 0 0 1 1 ...
 $ Pclass     : int  3 1 3 1 3 3 1 3 3 2 ...
 $ Name       : chr  "Braund, Mr. Owen Harris" "Cumings, Mrs. John Bradley (Florence Briggs Thayer)" "Heikkinen, Miss. Laina" "Futrelle, Mrs. Jacques Heath (Lily May Peel)" ...
 $ Sex        : chr  "male" "female" "female" "female" ...
 $ Age        : num  22 38 26 35 35 NA 54 2 27 14 ...
 $ SibSp      : int  1 1 0 1 0 0 0 3 0 1 ...
 $ Parch      : int  0 0 0 0 0 0 0 1 2 0 ...
 $ Ticket     : chr  "A/5 21171" "PC 17599" "STON/O2. 3101282" "113803" ...
 $ Fare       : num  7.25 71.28 7.92 53.1 8.05 ...
 $ Cabin      : chr  "" "C85" "" "C123" ...
 $ Embarked   : chr  "S" "C" "S" "S" ...
'data.frame':	418 obs. of  11 variables:
 $ PassengerId: int  892 893 894 895 896 897 898 899 900 901 ...
 $ Pclass     : int  3 3 2 3 3 3 3 2 3 3 ...
 $ Name       : chr  "Kelly, Mr. James" "Wilkes, Mrs. James (Ellen Needs)" "Myle

**Proporcion de los sobrevivientes**

In [ ]:
table(train$Survived)
prop.table(table(train$Survived))

**Primera prediccion (todos mueren)**

In [14]:
test$Survived<- rep(0,418)

submit<-data.frame(PassengerId=test$PassengerId,Survived=test$Survived)
write.csv(submit, file = "allDead.csv", row.names = FALSE)

**Porcentaje de supervivencia entre hombres y mujeres**

In [31]:
prop.table(table(train$Survived,train$Sex),2)

   
       female      male
  0 0.2579618 0.8110919
  1 0.7420382 0.1889081

Del total de los hombres murió el 81.1% y de las mujeres murió el 25.8%

**Prediccion 2: hombres mueren, mujeres viven**

In [35]:
test$Survived<- rep(0,418)
test$Survived[test$Sex=="female"]<-1
submit<-data.frame(PassengerId=test$PassengerId,Survived=test$Survived)
write.csv(submit, file = "allDead.csv", row.names = FALSE)